cpp/periodic_module/periodic_iterate.cpp

```diff
@@ -0,0 +1,241 @@
+      } else {
+        constructed_row.Insert(columns[i], batch[row][i]);
+      }
+    }
+
+    list_value.Append(mg::Value(std::move(constructed_row)));
+  }
+
+  params.Insert(kBatchInternalName, mg::Value(std::move(list_value)));
+
+  return params;
+}
+
+std::string ConstructFinalQuery(std::string running_query, std::string preffix_query) {
+  return fmt::format("{} {}", preffix_query, running_query);
+}
+
+void ExecuteRunningQuery(std::string running_query, std::vector<std::string> columns,
+                         std::vector<std::vector<mg::Value>> &batch) {
+  if (!batch.size()) {
+    return;
+  }
+
+  auto param_names = ExtractParamNames(columns, batch[0]);
+  auto query_columns = ConstructColumns(columns, batch[0]);
+
+  auto preffix_query = ConstructQueryPreffix(param_names);
+  auto final_query = ConstructFinalQuery(running_query, preffix_query);
+
+  auto query_params = ConstructParams(columns, batch);
+
+  mg::Client::Params session_params{.host = "localhost", .port = 7687};
+  auto client = mg::Client::Connect(session_params);
+  if (!client) {
+    throw std::runtime_error("Unable to connect to client!");
+  }
+  if (!client->Execute(final_query, query_params.AsConstMap())) {
+    throw std::runtime_error("Error while executing periodic iterate!");
+  }
+
+  client->DiscardAll();
+}
+
+void PeriodicIterate(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+  auto record = record_factory.NewRecord();
+
+  const auto input_query = std::string(arguments[0].ValueString());
+  const auto running_query = std::string(arguments[1].ValueString());
+  const auto config = arguments[2].ValueMap();
+
+  const auto batch_size_value = config.At(kConfigKeyBatchSize);
+
+  try {
+    if (batch_size_value.IsNull()) {
+      throw std::runtime_error(fmt::format("Configuration parameter {} is not set.", kConfigKeyBatchSize));
+    }
+    if (!batch_size_value.IsInt()) {
+      throw std::runtime_error("Batch size not provided as an integer in the periodic iterate configuration!");
+    }
+
+    const auto batch_size = batch_size_value.ValueInt();
+
+    if (batch_size <= 0) {
+      throw std::runtime_error("Batch size must be a non-negative number!");
+    }
+
+    mg::Client::Init();
+
+    mg::Client::Params params{.host = "localhost", .port = 7687};
+    auto client = mg::Client::Connect(params);
+
+    if (!client) {
+      throw std::runtime_error("Unable to connect to client!");
+    }
+
+    if (!client->Execute(input_query)) {
+      record.Insert(kReturnSuccess, false);
+      return;
+    }
+
+    auto columns = client->GetColumns();
+
+    std::vector<std::vector<mg::Value>> batch;
+    batch.reserve(batch_size);
+    int rows = 0;
+    while (const auto maybe_result = client->FetchOne()) {
```

Comment:
but it should exist since `const auto maybe_result = client->FetchOne()` will evaluate to bool

Reply:
that's why it's in the while condition

Reply:
shouldn't then the condition be:
`while (const auto maybe_result = client->FetchOne(); maybe_result)`

Reply:
But which type is then maybe_result?

Reply:
I think both `while (const auto maybe_result = client->FetchOne(); maybe_result)` and  while(const auto maybe_result = client->FetchOne()) are correct. @as51340  `std::optional<std::vector<Value>>` is type and when stream is empty it returns std::nullopt, that's why this will work correctly.

Reply:
it's optional, and then with the assignment in the condition it is evaluated like true or false.